osquery/killswitch/killswitch_refreshable_plugin.cpp

```diff
@@ -0,0 +1,53 @@
+#include <osquery/dispatcher.h>
+#include <osquery/flags.h>
+#include <osquery/killswitch.h>
+#include <osquery/killswitch/killswitch_refreshable_plugin.h>
+
+namespace osquery {
+
+FLAG(uint32,
+     killswitch_refresh_rate,
+     10,
+     "Refresh rate of killswitch in seconds");
+
+class KillswitchRefresher : public InternalRunnable {
+ public:
+  KillswitchRefresher(size_t update_interval)
+      : InternalRunnable("KillswitchRefreshRunner"),
+        update_interval_(update_interval) {}
+  /// A simple wait/interruptible lock.
+  void start() override {
+    while (!interrupted()) {
+      osquery::Killswitch::get().refresh();
+      pauseMilli(
+          std::chrono::milliseconds(std::chrono::seconds(update_interval_)));
+    }
+  }
+
+ private:
+  const size_t update_interval_;
+};
+
+Status KillswitchRefreshablePlugin::setUp() {
+  if (FLAGS_killswitch_refresh_rate > 0) {
+    Dispatcher::addService(
+        std::make_shared<KillswitchRefresher>(FLAGS_killswitch_refresh_rate));
+  }
+  return Status::success();
+}
+
+Status KillswitchRefreshablePlugin::call(const PluginRequest& request,
+                                         PluginResponse& response) {
+  auto action = request.find("action");
+  if (action == request.end()) {
+    return Status(1, "Config plugins require an action");
```

Comment:
You also could use `Status::failure("<your nice error message>");` here, if you like.